System/Posix/Files/Common.hsc

```diff
@@ -688,3 +718,277 @@ getFdPathVar (Fd fd) v =
+{-# WARNING fileBlockSizeX "fileBlockSizeX: not available on this platform" #-}
+fileBlockSizeX _ = error "fileBlockSizeX not available on this platform"
+{-# WARNING fileOwnerX "fileOwnerX: not available on this platform" #-}
+fileOwnerX _ = error "fileOwnerX not available on this platform"
+{-# WARNING fileGroupX "fileGroupX: not available on this platform" #-}
+fileGroupX _ = error "fileGroupX not available on this platform"
+{-# WARNING fileModeX "fileModeX: not available on this platform" #-}
+fileModeX _ = error "fileModeX not available on this platform"
+{-# WARNING fileIDX "fileIDX: not available on this platform" #-}
+fileIDX _ = error "fileIDX not available on this platform"
+{-# WARNING fileSizeX "fileSizeX: not available on this platform" #-}
+fileSizeX _ = error "fileSizeX not available on this platform"
+{-# WARNING fileBlocksX "fileBlocksX: not available on this platform" #-}
+fileBlocksX _ = error "fileBlocksX not available on this platform"
+{-# WARNING fileAttributesMaskX "fileAttributesMaskX: not available on this platform" #-}
+fileAttributesMaskX _ = error "fileAttributesMaskX not available on this platform"
+{-# WARNING fileAccessTimeHiResX "fileAccessTimeHiResX: not available on this platform" #-}
+fileAccessTimeHiResX _ = error "fileAccessTimeHiResX not available on this platform"
+{-# WARNING fileCreationTimeHiResX "fileCreationTimeHiResX: not available on this platform" #-}
+fileCreationTimeHiResX _ = error "fileCreationTimeHiResX not available on this platform"
+{-# WARNING fileStatusChangeTimeHiResX "fileStatusChangeTimeHiResX: not available on this platform" #-}
+fileStatusChangeTimeHiResX _ = error "fileStatusChangeTimeHiResX not available on this platform"
+{-# WARNING fileModificationTimeHiResX "fileModificationTimeHiResX: not available on this platform" #-}
+fileModificationTimeHiResX _ = error "fileModificationTimeHiResX not available on this platform"
+{-# WARNING fileCompressedX "fileCompressedX: not available on this platform" #-}
+fileCompressedX _ = error "fileCompressedX not available on this platform"
+{-# WARNING fileImmutableX "fileImmutableX: not available on this platform" #-}
+fileImmutableX _ = error "fileImmutableX not available on this platform"
+#endif
+
+data {-# CTYPE "struct statx" #-} CStatx
+
+#ifdef HAVE_STATX
+foreign import ccall unsafe "statx"
+   c_statx :: CInt -> CFilePath -> CInt -> CInt -> Ptr CStatx -> IO CInt
+#endif
+
+
+getExtendedFileStatus_ :: Maybe Fd  -- ^ Optional directory file descriptor
+                       -> CString   -- ^ Pathname to open
+                       -> StatxFlags
+                       -> StatxMask
+                       -> IO ExtendedFileStatus
+#ifdef HAVE_STATX
+getExtendedFileStatus_ fdMay str StatxFlags{..} StatxMask{..} = do
+  fp <- mallocForeignPtrBytes (#const sizeof(struct statx))
+  withForeignPtr fp $ \p ->
+      throwErrnoIfMinus1Retry_ "getExtendedFileStatus_" (c_statx c_fd str flags masks p)
```

Comment:
Is `throwErrnoIfMinus1Retry_` good here?

Reply:
Are you asking whether it is *necessary* (whether `statx(2)` is interruptible) or whether it is the right thing to do (`statx(2)` is interruptible , but we shouldn't retry)?

The manpage for `statx(2)` does not mention `EINTR`. Plausibly it is not in fact interruptible (caught signals don't interrupt the call).